src/main/java/com/amazon/opendistroforelasticsearch/security/support/wildcard/Wildcard.java

```diff
@@ -0,0 +1,70 @@
+package com.amazon.opendistroforelasticsearch.security.support.wildcard;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.stream.Collectors;
+
+// Case-sensitivity and other options are defined by concrete Wildcard implementation
+public interface Wildcard {
+    boolean matches(String candidate);
+
+    default boolean matchesAny(Collection<String> candidates) {
+        return candidates.stream().anyMatch(this::matches);
+    }
+
+    default boolean matchesAny(String[] candidates) {
+        return Arrays.stream(candidates).anyMatch(this::matches);
+    }
+
+    // TODO: make serializable, hashable etc.
+    Wildcard ANY = new Wildcard() {
+        @Override
+        public boolean matches(String candidate) {
+            return true;
+        }
+    };
+
+    // TODO: make serializable, hashable etc.
+    Wildcard NONE = new Wildcard() {
+        @Override
+        public boolean matches(String candidate) {
+            return false;
+        }
+    };
+
+    // This may in future use more optimized techniques to combine multiple wildcards in a single automaton
+    static Wildcard caseSensitiveAny(Collection<String> patterns) {
```

Comment:
use flag for case sensitivity: `any(Collection<String> patterns, boolean isCaseSensitive)`

Reply:
I believe case sensitive is a property of pattern not of matching. Much like JDK Pattern class.

Reply:
It is a property of `WildcardMatcher` and there is no difference whether it is designated by the method name `caseSensitiveAny(patterns)` or by the flag `any(pattern, true)`.

Reply:
Or you mean the factory method, yeah makes sense